AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationStrategies/MagneticAllocation/MagneticAllocation.py

```diff
@@ -281,21 +303,26 @@ def _try_put_all_candidates(
         :rtype: tuple[int, int]
         """
 
+        results_evaluation = joblib.Parallel(n_jobs=-1, verbose=0)(
+            joblib.delayed(self._try_put_position)(
+                center_position, size, position_from
+            )
+            for center_position in center_positions
+        )
+
+        # find best score
         best_position = None
         best_score = -float("inf")
+        for cnt, result_evaluation in enumerate(results_evaluation):
+            if (result_evaluation is not None) and (
+                result_evaluation > best_score
+            ):
+                best_score = result_evaluation
+                best_position = center_positions[cnt]
 
-        for center_position in center_positions:
-            # if hitting with other word...
-            if self._is_hitting_other_words(center_position, size):
-                # ...skip this position
-                continue
-
-            score = self._evaluate_position(position_from, center_position)
-
-            if score > best_score:
-                # best score updated
-                best_position = center_position
-                best_score = score
+        # guard
+        if best_position is None:
+            raise Exception("No position found")
 
         return best_position
 
```

Comment:
function _is_hitting_other_words' s size argument is weird. [int, int]->tuple[int, int]?

Reply:
fixed